app/plugins/newgrounds.ts

```diff
@@ -62,18 +76,62 @@ export class NewgroundsPlugin extends BaseSitePlugin {
     }
 
     async checkFileDownload(): Promise<I.Media> {
+        // old comment:
         // Only check file downloads for art; other media has to be treated specially.
-        return this.getArtMedia();
+
+        // new comment: check 'em all!
+
+        let downloadable = this.getArtMedia();
+        if(downloadable) {
+            return downloadable;
+        }
+
+        downloadable = this.getVideoMedia();
+        if(downloadable) {
+            return downloadable;
+        }
```

Comment:
Doesn't this trigger the "press play to download videos" dialog? Even if a user doesn't have autodownload on, they'll see that dialog just visiting the page as I use this to determine whether a piece of media has already been downloaded.

Reply:
It might; I had the "press play" dialog trigger at least once when I first visited a movie page.

The problem I was trying to solve when I expanded `checkFileDownload()` was that it seemed to be preventing auto-download of other-than-art files.  The logic starting [about here in page.tsx](https://github.com/Simon-Tesla/RaccoonyWebEx/blob/master/app/ui/page/page.tsx#L228) wasn't making it to the "if auto-download is on, then do the download" part (page.tsx line 245).  I decided that was because `checkFileDownload()`, as was, wasn't returning anything for `media` when it was called on a movie or audio file, so the stuff inside the `if(media)` (page.tsx line 229) never ran.

I thought this function's job was to tell you if it was *possible* to download something from a particular page, which is why I had it run the "get media" functions for both movie and music submissions, just like it was already doing for art submissions.  I didn't think it was supposed to tell you if you had *already* downloaded something from a certain page.

Reply:
The `checkFileDownload` method is about determining if the user has downloaded the file already and to download it if the user has auto-download turned on. It has nothing to do with the decision about whether it's possible to download; that's what `hasMedia` is for (though by default the implementation for `hasMedia` is to call `getMedia` and see if it returns a non-null response. 

I added `checkFileDownload` as part of the original Newgrounds support specifically to suppress that dialog, as it lets the code distinguish between the automatic check and a user-initiated download. When I originally wrote it, it looked impossible to download video without user interaction, so it didn't make sense to do the check since it'd both not work and show a confusing dialog without user interaction. I forgot to add audio support when I merged in your change for that, which is an oversight, but I think video auto-download likely would remain unsupported. At the minimum, we need to make sure it doesn't show the dialog when it's called.

Reply:
(I should really add some documentation to base.ts now that it's somewhat more complex)